lgc/builder/BuilderImpl.cpp

```diff
@@ -323,6 +323,29 @@ BranchInst *BuilderImplBase::createIf(Value *condition, bool wantElse, const Twi
   return branch;
 }
 
+// Permissively test two instructions are the same.
+static bool instructionsIdentical(Instruction *lhs, Instruction *rhs) {
+  if (lhs->isIdenticalTo(rhs))
+    return true;
+
+  if (lhs->isSameOperationAs(rhs)) {
```

Comment:
nit: prefer early returns

Reply:
I am not sure I understand as this is an early return?

Reply:
We could do something like:
```c++
if (!lhs->isSameOperationAs(rhs))
  return true;
  
for (unsigned idx ...) {

}
return true;
```

and reduce one level of nesting. See https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code and https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html.

Since the function is not too long, I think either format is fine, thus 'nit'.

Reply:
Thanks for the clarification, I was focusing on the first line of the diff not the last one.